Assets/Scripts/Common/AStateContext.cs

```diff
@@ -11,10 +11,18 @@ public abstract class AStateContext : MonoBehaviour
 {
     private IState _state;
 
-    protected IState State
+    public IState State
     {
         get { return this._state; }
-        set { this.SetState(value); }
+    }
+
+    /// <summary>
+    /// Returns true if the current state is the given state.
+    /// </summary>
+    public bool IsInState<StateT>()
+      where StateT : IState
+    {
+        return this.State.GetType() == typeof(StateT);
```

Comment:
I used this in a previous version, but I don't use it anymore. However it might still be useful in the future. Should we leave it here or add it back later if necessary?